lib/cim/cim-adapter.ts

```diff
@@ -0,0 +1,49 @@
+import {PimClass} from "../pim/model";
+import {CoreResourceReader} from "../core";
+import {IriProvider} from "./iri-provider";
+
+export interface CimAdapter {
+
+    /**
+     * Resources returned by the adapter are on the PIM level, therefore the PIM IRIs are needed to interconnect them.
+     * Also, the IRIs need to be predictable and revertible so the user is able to get CIM iri from PIM iri.
+     * @param iriProvider
+     */
+    setIriProvider(iriProvider: IriProvider): void;
```

Comment:
Is there a default provider? If not what happen if user does not call this method? I mean there is nothing forcing him to call it, or even suggesting it. I would move it to the implementation's builder of factory class.